main.go

```diff
@@ -460,25 +460,37 @@ func main() {
 		hardcodedLabels := []string{"group", "handler"}
 		instrumenter := server.NewInstrumentedHandlerFactory(reg, hardcodedLabels)
 
-		tenantIDs := map[string]string{}
-		authorizers := map[string]rbac.Authorizer{}
-		var rateLimits []ratelimit.Config
-		oidcTenants := map[string]struct{}{}
-		// registrationRetryCount used by authenticator providers to count
-		// registration failures per tenant.
-		registerTenantsFailingMetric := authentication.RegisterTenantsFailingMetric(reg)
-		pm := authentication.NewProviderManager(logger, registerTenantsFailingMetric)
+		var (
+			tenantIDs   = map[string]string{}
+			authorizers = map[string]rbac.Authorizer{}
+			oidcTenants = map[string]struct{}{}
+
+			rateLimits []ratelimit.Config
+			// registrationRetryCount used by authenticator providers to count
+			// registration failures per tenant.
+			registerTenantsFailingMetric = authentication.RegisterTenantsFailingMetric(reg)
+			pm                           = authentication.NewProviderManager(logger, registerTenantsFailingMetric)
+		)
 
 		r.Group(func(r chi.Router) {
-			// registeredAuthNRoutes is used to avoid double register the same pattern.
+			// Set up common middleware before mounting authN routes.
+			for _, t := range tenantsCfg.Tenants {
+				tenantIDs[t.Name] = t.ID
+			}
+
+			r.Use(authentication.WithTenant)
+			r.Use(authentication.WithTenantID(tenantIDs))
+			r.Use(authentication.WithAccessToken())
+			r.MethodNotAllowed(blockNonDefinedMethods())
+
+			// registeredAuthNRoutes is used to avoid double registration of the same pattern.
 			var regMtx sync.RWMutex
 			registeredAuthNRoutes := make(map[string]struct{})
 			for _, t := range tenantsCfg.Tenants {
 				if t == nil {
```

Comment:
not sure I get why we check here if `t == nil` - would we still need this? if yes, would it be needed also [here](https://github.com/observatorium/api/pull/325/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261R477-R479)?

Reply:
I think you are correct, if there is a risk for panic we should be checking `t` and if not we can remove both checks. We appear to have a similar situation with the rate limits where `rl` is a pointer but we don't do a nil check.

Reply:
Good catch, let me remove it then :broom: